compilation error due to `std::remove_reference`

I met compilation error on EASTL 3.21.12. 

```
In file included from include/EASTL/type_traits.h:1033,
                 from include/EASTL/internal/move_help.h:16,
                 from include/EASTL/functional.h:11,
                 from include/EASTL/internal/fixed_pool.h:28,
                 from source/fixed_pool.cpp:6:
include/EASTL/internal/type_transformations.h:500:74: error: 'remove_reference' in namespace 'std' does not name a template type
  500 |                 auto try_add_pointer(int) -> type_identity<typename std::remove_reference<T>::type*>;
      |                                                                          ^~~~~~~~~~~~~~~~
include/EASTL/internal/type_transformations.h:500:90: error: expected template-argument before '<' token
  500 |                 auto try_add_pointer(int) -> type_identity<typename std::remove_reference<T>::type*>;
```

Environments as follows:
- OS: Debian stretch
- Compiler: gcc 12.1.0
- EASTL: 3.21.12

I succeeded compilation on EASTL 3.17.03.

I think it is caused by modification of include/EASTL/internal/type_transformations.h.

https://github.com/electronicarts/EASTL/commit/1aa784643081404783ce6494eb2fcaba99d8f6a5#diff-6b653b7c05d254bf37fddd53af0feb52d423005e81fadbae783b1d8e6054d74e

I thknk `try_add_pointer(int)` should use `eastl::remove_reference`?

```
template <typename T>
auto try_add_pointer(int) -> type_identity<typename eastl::remove_reference<T>::type*>;
```

--- a/include/EASTL/array.h
+++ b/include/EASTL/array.h
@@ -617,7 +617,7 @@ namespace eastl
 	struct tuple_element {};
 
 	template<size_t I, typename T, size_t N>
-	struct tuple_element<I, T, N, std::enable_if_t<(I < N)>> {
+	struct tuple_element<I, T, N, eastl::enable_if_t<(I < N)>> {
 		using type = T;
 	};
 	}

--- a/include/EASTL/internal/type_transformations.h
+++ b/include/EASTL/internal/type_transformations.h
@@ -497,7 +497,7 @@ namespace eastl
 	namespace internal
 	{
 		template <typename T>
-		auto try_add_pointer(int) -> type_identity<typename std::remove_reference<T>::type*>;
+		auto try_add_pointer(int) -> type_identity<typename eastl::remove_reference<T>::type*>;
 		template <typename T>
 		auto try_add_pointer(...) -> type_identity<T>;
 	}

--- a/include/EASTL/span.h
+++ b/include/EASTL/span.h
@@ -63,7 +63,7 @@ namespace eastl
 		struct is_eastl_array : public eastl::false_type {};
 
 		template<typename T, size_t N>
-		struct is_eastl_array<std::array<T, N>, T> : public eastl::true_type {};
+		struct is_eastl_array<eastl::array<T, N>, T> : public eastl::true_type {};
 	}
 
 	template <typename T, size_t Extent = eastl::dynamic_extent>

